tests/test_ideogram.py

```diff
@@ -0,0 +1,311 @@
+# define any custom strings here (e.g., if you use
+# a particular CSS selector a lot, assign it to a
+# variable)
+
+
+# Demo app tests
+
+@init_demo_app(APP_NAME)
+def test_click_app_name_from_gallery(dash_threaded, selenium):
+    """Test that clicking on the given app goes to the expected URL."""
+    assert selenium.current_url.replace('http://localhost:8050', '').strip('/') == \
+        'dash-bio/{}'.format(APP_NAME)
+
+
+# below, write tests for initial conditions; they will most likely
+# make use of wait_for_text_to_equal and other similar functions
+
+# Component tests
+
+# for React components, we need to define a way to interact with the
+# props directly (instead of through a graph component); to that end,
+# we define a callback method that defines how exactly a prop is
+# updated
+# this callback will be used in the simple test app, which consists of
+# the component, a single button, and two inputs
+
+
+def ideogram_test_props_callback(
+        nclicks,
+        prop_name,
+        prop_value,
+        prop_type=None
+):
+    """This function is the code of a callback which is triggered by
+    the button on the simple app used in the test.
+    :param nclicks (int): The n_clicks value of the button in the
+                          simple app
+    :param prop_name (string): The name of the property that is to be
+                               modified
+    :param prop_value (string): The value that is to be assigned to the
+                                prop defined by prop_name.
+    :prop_type (string): One of the predefined types in PROP_TYPES.
+    :return: The value that is to be assigned to the prop defined by
+             prop_name, after casting it to the correct type.
+    """
+    answer = None
+
+    if prop_type == 'dict':
+        answer = {}
+
+    # avoid triggering this callback when the button is first created
+    if nclicks is not None:
+        # cast the string representation of the desired prop value
+        # into the appropriate type
+        if prop_type in PROP_TYPES:
+            prop_value = PROP_TYPES[prop_type](prop_value)
+        answer = prop_value
+
+    return answer
+
+
+# below, write tests for changing the props of the React component
+# (following this basic structure)
+
+BASIC_PROPS = {
+    'organism': 'human',
+    'dataDir': 'https://unpkg.com/ideogram@1.3.0/dist/data/bands/native/'
+}
+
+
+def test_chr_height(dash_threaded, selenium):
+    """The pixel height of the tallest chromosome in the ideogram."""
+
+    prop_type = 'float'
+
+    def assert_callback(prop_value, nclicks, input_value):
+        answer = ''
+        if nclicks is not None:
+            answer = FAIL
+            if PROP_TYPES[prop_type](input_value) == prop_value:
+                answer = PASS
+        return answer
+
+    template_test_component_single_prop(
+        dash_threaded,
+        selenium,
+        APP_NAME,
+        assert_callback,
+        ideogram_test_props_callback,
+        "chrHeight",
```

Comment:
Noticing a mix of single and double quotes... Are these double quotes because these strings get displayed in the UI?